lib/xmerl/src/xmerl.erl

```diff
@@ -53,6 +52,10 @@ export(Content,Callback)
 
 Equivalent to [export(Data, Callback, [])](`export/3`).
 """.
+-spec export(Content, Callback) -> ExportedFormat when
+    Content :: [#xmlText{}|#xmlElement{}|#xmlPI{}|#xmlComment{}|#xmlDecl{}],
```

Comment:
I know very little about xml (and the xmerl app), and maybe I am overly abstract here,
but would it not be a good idea to introduce type(s) for these records (and then use these in the spec)?
That is:

-type text() :: #xmlText{}.
-type element() :: #xmlElement{}.
-type pi() :: #xmlPI{}.
-type comment() :: #xmlComment{}.
-type decl() :: #xmlDecl{}.

It would then be possible to document and explain these types (records).

Reply:
I did it this way like you mentioned of having a new type. but I see there are some more different date type in some other functions. Then I deleted my type definition and put all them in a raw way.